src/components/OrderModal.tsx

```diff
@@ -38,7 +39,13 @@ const OrderModal = ({ query, onDrawerOpen }: ModalProps) => {
         </ModalBody>
         <ModalFooter>
           {success ? (
-            <ActionButton square onClick={onClose}>
+            <ActionButton
+              square
+              onClick={() => {
+                onClose();
+                clearCart();
```

Comment:
you'll need to move this somewhere else, maybe the useEffect? it won't run if the modal's closed another way like Esc, closebutton